SWIPs/swip-adaptive-node-protocol.md

```diff
@@ -0,0 +1,131 @@
+---
+SWIP: 12
+title: Adaptive nodes capabilities protocol
+author: Louis Holbrook <dev@holbrook.no> (https://holbrook.no), Tim Bansemer (editorial) @FantasticoFox <tim@ethswarm.org>
+status: Draft
+type: Track Networking
+created: 2019-07-11
+---
+
+## Abstract
+
+Adaptive node operation means a node should be able to change the capabilities it has according to changes in available resources.
+
+## Motivation
+
+The peers of the node may need to adjust their behavior towards the node when these capabilities change. Therefore, a protocol is required to inform the peers of changes in capabilities.
+
+## Specification
+
+Upon initial connection, capabilities can be transmitted already within the handshake message.
+
+Capabilities are combinational. Some combination may make more sense than other. Interpreting which combinations are feasible and which are not should be handled by respective nodes, and not be hardwired in the protocol.
+
+For this reason, a bit vector seems the most logical choice. The bit vector can represent the adaptive node operation state within the node, and be embedded into the protocol message as-is.
```

Comment:
"seems" is a bit a of a weasel word in a spec. Better to just state what it does